bin/switchy

```diff
@@ -0,0 +1,116 @@
+@click.option('--interactive/--non-interactive', default=False, help='Whether to jump into an interactive session after setting up the call originator')
+@click.option('--debug/--no-debug', default=False, help='Whether to enable debugging')
+@click.option('--app', default='Bert', help='Switchy application to execute (see list-apps command to list available apps)')
+@click.option('--metrics-file', default=None, help='Store metrics at the given file location')
+def run(slaves, proxy, profile, rate, limit, max_offered, duration, interactive, debug, app, metrics_file):
+    logging.basicConfig()
+
+    # Check if the specified (or default) app is valid
+    switchy.apps.load()
+    appdata = switchy.apps.get(app)
+    if not appdata:
+        raise click.ClickException('Unknown app {}. Use list-apps command to list available apps'.format(app))
+
+    # TODO: get_originator() receives an apps tuple (defaults to Bert) to select the application
+    # we should accept --app multi argument list to set multiple apps
+    o = switchy.get_originator(slaves, apps=(appdata.cls, ))
+
+    # Set basic originator settings
+    if rate:
+        o.rate = int(rate)
+    if limit:
+        o.limit = int(limit)
+    if max_offered:
+        o.max_offered = int(max_offered)
+    if duration:
+        o.duration = int(duration)
+
+    # Prepare the originate string for each slave
+    # depending on the profile name and network settings
+    # configured for that profile in that particular slave
+    p = re.compile('.+?BIND-URL\s+?.+?@(.+?):(\d+).+?\s+', re.IGNORECASE | re.DOTALL)
+    for c in o.pool.clients:
+        status = c.client.api('sofia status profile {}'.format(profile)).getBody()
+        m = p.match(status)
+        if not m:
+            raise click.ClickException('Slave {} does not have a profile named \'{}\' running'.format(c.host, profile))
+        ip = m.group(1)
+        port = m.group(2)
+        # The originate cmd must route the call back to us using the specified proxy (the device under test)
+        click.echo('Slave {} SIP address is at {}:{}'.format(c.host, ip, port))
+        c.set_orig_cmd(dest_url='switchy@{}:{}'.format(ip, port), profile=profile, app_name='park', proxy='{}'.format(proxy))
+
+    click.echo('---------------------------------------------------------------')
+    click.echo('Starting load test for server {} at {}cps using {} load slaves'.format(proxy, o.rate, len(slaves)))
+    click.echo('---------------------------------------------------------------')
+    click.echo(str(o))
+    if interactive:
+        import IPython
+        IPython.embed()
+        o.shutdown()
+        print o
+    else:
+        o.start()
+        while o.state != 'STOPPED':
+            try:
+                time.sleep(1)
+                print o
+            except KeyboardInterrupt:
+                o.shutdown()
+                print o
+
+    active_calls = o.count_calls()
```

Comment:
It could. That was my first impulse but then decided to do it this way to be consistent, otherwise you can end up printing a message 'Waiting on 0 active calls to finish', which makes no sense. Unfortunately the not-so BDFL didn't like assignments in if conditions so I can't do:

```python
while (active_calls = o.count_calls()) > 0:
```

To avoid calling count_calls() before and within the loop.

Just for fun, I ended up changing it to use an infinite loop breaking only when active_calls reaches 0